### PR TITLE
feat:added toogle password function

### PR DESCRIPTION
**Pull request by jatingodnani**
#558 @deepakprabhakara @devkiran check my pr.added toogle password

### CODE REVIEW

**Comment by deepakprabhakara** [comment]
@jatingodnani Can you please format the files, `npm run format`

**Comment by jatingodnani** [comment]
done @devkiran

File: components/auth/Join.tsx [hunk 1]
```diff
@@ -9,11 +10,14 @@ import toast from 'react-hot-toast';
 import type { ApiResponse } from 'types';
 import * as Yup from 'yup';
 import Link from 'next/link';
-
+import { MdVisibility, MdVisibilityOff } from 'react-icons/md';
```
Comment (devkiran) [inline components/auth/Join.tsx:13]: Can we use `heroicons` instead? We'll remove the react-icons from the project.

File: components/auth/Join.tsx [hunk 2]
```diff
@@ -9,11 +10,14 @@ import toast from 'react-hot-toast';
...
 const Join = () => {
   const router = useRouter();
   const { t } = useTranslation('common');
-
+  const [isPasswordVisible, setIsPasswordVisible] = useState<boolean>(true);
+  const handlepasswordvisibility = () => {
```
Comment (devkiran) [inline components/auth/Join.tsx:18]:
```suggestion
  const hadlePasswordVisibility = () => {
```

File: components/auth/Join.tsx [hunk 3]
```diff
@@ -91,15 +95,27 @@ const Join = () => {
...
+            type={isPasswordVisible ? 'text' : 'password'}
+            label={t('password')}
+            name="password"
+            placeholder={t('password')}
+            value={formik.values.password}
+            error={formik.touched.password ? formik.errors.password : undefined}
+            onChange={formik.handleChange}
+          />
+          <button
+            onClick={handlepasswordvisibility}
```
Comment (devkiran) [inline components/auth/Join.tsx:109]:
```suggestion
            onClick={hadlePasswordVisibility}
```

Comment (devkiran) [reply components/auth/Join.tsx:109]: Add property `type='button'` to the button, otherwise it will submit the form when click the icon.

**Review by devkiran — CHANGES_REQUESTED** [review]

File: components/auth/Join.tsx (hunk 2 above)
Comment (jatingodnani) [reply components/auth/Join.tsx:18]: what is the problem in this

Comment (devkiran) [reply components/auth/Join.tsx:18]: Please use camelCase for method name.

Comment (jatingodnani) [reply components/auth/Join.tsx:18]: Ok

**Comment by jatingodnani** [comment]
@devkiran  done

File: package.json [hunk 4]
```diff
@@ -23,7 +23,7 @@
     "@boxyhq/metrics": "0.2.5",
     "@boxyhq/react-ui": "3.3.13",
     "@boxyhq/saml-jackson": "1.13.2",
-    "@heroicons/react": "2.0.18",
+    "@heroicons/react": "^2.0.18",
     "@hookform/resolvers": "3.3.1",
```
Comment (devkiran) [inline package.json:27]: Heroicon is already installed in the project, can you revert the changes to package.json/package-lock.json?

**Review by devkiran — CHANGES_REQUESTED** [review]
Can you please look at the following issues?

- Eye icon is not visible in the light mode
- Can we reduce the size of the icon? Maybe from 6 to 5. Looks a bit large at the moment.
- Can we add the similar feature to JoinWithInvitation as well?
- Maybe you can move this to a separate component and use it login/join pages? (Eg: TogglePasswordVisibility.tsx)

**Comment by jatingodnani** [comment]
@devkiran check

**Comment by jatingodnani** [comment]
@devkiran is their any problem in my code

**Review by devkiran — APPROVED** [review]

**Comment by devkiran** [comment]
@jatingodnani Thanks for the PR.

**Comment by devkiran** [comment]
@jatingodnani Thanks for the PR.

**Pull request merged**